File: ang/src/app/dto/error.dto.ts
import {Response} from '@angular/http';

/**
 *  DTO représentatif d'une erreur interne.
 */
export class ErrorDTO {
  /**
   *  Message de notification.
   */
  public notification: string;

  /**
   *  Réponse envoyée par le serveur.
   */
  public response: Response;
}

File: src/com/akasoft/poneyrox/dto/ErrorDTO.java
package com.akasoft.poneyrox.dto;

import com.akasoft.poneyrox.exceptions.AbstractException;

/**
 *  Erreur.
 *  DTO représentatif d'une erreur levée dans l'application.
 */
public class ErrorDTO {
    /**
     *  Notification.
     */
    private String notification;

    /**
     *  Constructeur.
     *  @param exception Exception représentée.
     */
    public ErrorDTO(AbstractException exception) {
        this.notification = exception.getMessage();
    }
}
